llvm/test/CodeGen/LoongArch/lasx/ir-instruction/absd.ll

```diff
@@ -125,202 +128,430 @@ define <4 x i64> @xvabsd_du(<4 x i64> %a, <4 x i64> %b) {
+; CHECK-NEXT:    xvsub.w $xr0, $xr0, $xr2
+; CHECK-NEXT:    ret
+  %cmp = icmp sgt <8 x i32> %a, %b
+  %ab = select <8 x i1> %cmp, <8 x i32> %a, <8 x i32> %b
+  %ba = select <8 x i1> %cmp, <8 x i32> %b, <8 x i32> %a
+  %sub = sub <8 x i32> %ab, %ba
+  ret <8 x i32> %sub
+}
+
+define <4 x i64> @xvabsd_d_select(<4 x i64> %a, <4 x i64> %b) nounwind {
+; CHECK-LABEL: xvabsd_d_select:
+; CHECK:       # %bb.0:
+; CHECK-NEXT:    xvmin.d $xr2, $xr0, $xr1
+; CHECK-NEXT:    xvmax.d $xr0, $xr0, $xr1
+; CHECK-NEXT:    xvsub.d $xr0, $xr0, $xr2
+; CHECK-NEXT:    ret
+  %cmp = icmp sge <4 x i64> %a, %b
+  %ab = select <4 x i1> %cmp, <4 x i64> %a, <4 x i64> %b
+  %ba = select <4 x i1> %cmp, <4 x i64> %b, <4 x i64> %a
+  %sub = sub <4 x i64> %ab, %ba
+  ret <4 x i64> %sub
+}
+
+define <32 x i8> @xvabsd_bu_select(<32 x i8> %a, <32 x i8> %b) nounwind {
+; CHECK-LABEL: xvabsd_bu_select:
+; CHECK:       # %bb.0:
+; CHECK-NEXT:    xvmin.bu $xr2, $xr0, $xr1
+; CHECK-NEXT:    xvmax.bu $xr0, $xr0, $xr1
+; CHECK-NEXT:    xvsub.b $xr0, $xr0, $xr2
+; CHECK-NEXT:    ret
+  %cmp = icmp ult <32 x i8> %a, %b
+  %ab = select <32 x i1> %cmp, <32 x i8> %a, <32 x i8> %b
+  %ba = select <32 x i1> %cmp, <32 x i8> %b, <32 x i8> %a
+  %sub = sub <32 x i8> %ba, %ab
+  ret <32 x i8> %sub
+}
+
+define <16 x i16> @xvabsd_hu_select(<16 x i16> %a, <16 x i16> %b) nounwind {
+; CHECK-LABEL: xvabsd_hu_select:
+; CHECK:       # %bb.0:
+; CHECK-NEXT:    xvmin.hu $xr2, $xr0, $xr1
+; CHECK-NEXT:    xvmax.hu $xr0, $xr0, $xr1
+; CHECK-NEXT:    xvsub.h $xr0, $xr0, $xr2
+; CHECK-NEXT:    ret
+  %cmp = icmp ule <16 x i16> %a, %b
+  %ab = select <16 x i1> %cmp, <16 x i16> %a, <16 x i16> %b
+  %ba = select <16 x i1> %cmp, <16 x i16> %b, <16 x i16> %a
+  %sub = sub <16 x i16> %ba, %ab
+  ret <16 x i16> %sub
+}
+
+define <8 x i32> @xvabsd_wu_select(<8 x i32> %a, <8 x i32> %b) nounwind {
+; CHECK-LABEL: xvabsd_wu_select:
+; CHECK:       # %bb.0:
+; CHECK-NEXT:    xvmin.wu $xr2, $xr0, $xr1
+; CHECK-NEXT:    xvmax.wu $xr0, $xr0, $xr1
+; CHECK-NEXT:    xvsub.w $xr0, $xr0, $xr2
+; CHECK-NEXT:    ret
+  %cmp = icmp ugt <8 x i32> %a, %b
+  %ab = select <8 x i1> %cmp, <8 x i32> %a, <8 x i32> %b
+  %ba = select <8 x i1> %cmp, <8 x i32> %b, <8 x i32> %a
+  %sub = sub <8 x i32> %ab, %ba
+  ret <8 x i32> %sub
+}
+
+define <4 x i64> @xvabsd_du_select(<4 x i64> %a, <4 x i64> %b) nounwind {
+; CHECK-LABEL: xvabsd_du_select:
+; CHECK:       # %bb.0:
+; CHECK-NEXT:    xvmin.du $xr2, $xr0, $xr1
+; CHECK-NEXT:    xvmax.du $xr0, $xr0, $xr1
+; CHECK-NEXT:    xvsub.d $xr0, $xr0, $xr2
+; CHECK-NEXT:    ret
+  %cmp = icmp uge <4 x i64> %a, %b
+  %ab = select <4 x i1> %cmp, <4 x i64> %a, <4 x i64> %b
+  %ba = select <4 x i1> %cmp, <4 x i64> %b, <4 x i64> %a
+  %sub = sub <4 x i64> %ab, %ba
+  ret <4 x i64> %sub
+}
+
+declare <32 x i8> @llvm.abs.v16i8(<32 x i8>, i1)
 
-declare <16 x i16> @llvm.abs.v16i16(<16 x i16>, i1)
-declare <32 x i16> @llvm.abs.v32i16(<32 x i16>, i1)
+declare <16 x i16> @llvm.abs.v8i16(<16 x i16>, i1)
```

Comment:
Why the intrinsic suffix is different from the type of return value and the first argument?